### DISCUSSION TITLE
Does MultiPar use CRC32 checksums?

### DISCUSSION THREAD

**Discussion by gareth41** [Yutaka-Sawada/MultiPar#100] [Q&A]
I'm just wondering how MultiPar verifies the integrity of the files before deciding which are corrupt and need repairing, is it using CRC32 checksums or SHA or similar?  Thanks in advance

**Comment by Yutaka-Sawada** [comment]
MultiPar is a parchive tool. There is a [Wiki page for parchive](https://en.wikipedia.org/wiki/Parchive). There is a page for [details of PAR2 file format](https://parchive.sourceforge.net/docs/specifications/parity-volume-spec/article-spec.html) also. It uses both CRC-32 and MD5 to detect damage of a file.
